src/main/java/io/cdap/plugin/snowflake/source/batch/SnowflakeMapToRecordTransformer.java

```diff
@@ -83,7 +84,8 @@ private Object convertValue(String fieldName, String value, Schema fieldSchema)
         case TIME_MICROS:
           return TimeUnit.NANOSECONDS.toMicros(LocalTime.parse(value).toNanoOfDay());
         case DECIMAL:
-          return new BigDecimal(value).setScale(fieldSchema.getScale()).unscaledValue().toByteArray();
+            return new BigDecimal(value).setScale(fieldSchema.getScale(),
+                                                  RoundingMode.HALF_EVEN).unscaledValue().toByteArray();
```

Comment:
why not `RoundingMode.HALF_UP`, which rounds towards the "nearest neighbor" unless both neighbors are equidistant?

Reply:
Its reference is taken from database plugins which uses HALF_EVEN. https://github.com/data-integrations/database-plugins/blob/81c88533143a4aecb10a157e964930cbac006e26/postgresql-plugin/src/main/java/io/cdap/plugin/postgres/PostgresDBRecord.java#L104.

There is minute difference between the both RoundingModes that too in case of .5, to make it common across plugins I chose this.